### ISSUE TITLE
Bug in trigfun "trunc"

### ISSUE THREAD

**Issue by trefethen** [chebfun/chebfun#2050]
The `'trunc'` option combined with `'trig'` should construct a finite trigonometric series with a specified number of terms.  It's doing it wrong, presumably because of the effect discussed in the footnote on p. C561 of the trigfun paper by Wright, Javed, Montanelli and Trefethen.  

I need this to work for the "Eight shades of rational approximation on an interval" paper I'm currently drafting with @mhsnjvd and @sfilip.  With that in mind I've taken the liberty of assigning this issue to you, @mhsnjvd.  Perhaps you can fix it quickly and @sfilip can review the result?

To illustrate, try this code:
```
f = chebfun('exp(sin(t))',[0,2*pi]);
tic, p = chebfun(f,'trunc',11,'trig'); toc
plot(f,'b',p,'r')
```
which produces the figure below.  The two curves should be more or less the same, but one is shifted to the wrong location.  On [-pi,pi] it would have come out fine.
![bug](https://cloud.githubusercontent.com/assets/3601488/22405261/ee0c12ce-e637-11e6-9bca-574c9a516480.png)

**Comment by trefethen** [comment]
Let me add a remark about another matter related to trig coefficients that could perhaps be sorted out in the same pull request.  If you look in section 11.7 of the _Chebfun Guide_, you'll see that we offer the alternative options `c = trigcoeffs(f)` (coeffs of a complex exponential series) and `[a,b] = trigcoeffs(f)` (coeffs of cosine and sine series).  But `help trigcoeffs` does not reveal the latter.  Could comments be added to `help trigcoeffs`?

Note at the end of that section the comments about `trigcoeffs(f)` vs. `f.coeffs`.  Plenty to think about here.

**Comment by gradywright** [comment]
It appears that `help trigcoeffs` gives the help text for `@trigtech/trigcoeffs` rather than `chebfun/trigcoeffs`.  When you type  `help chebfun/trigcoeffs` you will see the `[a,b]` output described.  I'm not sure why the `@trigtech` help text appears.

**Comment by trefethen** [comment]
I am afraid trigfun now has a new bug that I suspect was introduced by this fix.  @mhsnjvd, can you take a look?  And once it is sorted out add a test or two?  

Here's the bug:
```
>> f = chebfun(exp(-10*x.^2),[-pi,pi],'trig');
Warning: Function not resolved using 65536 pts. Have you tried a non-trig representation? 
> In chebfun/constructor>constructorNoSplit (line 120)
  In chebfun/constructor (line 63)
  In chebfun (line 219) 
>> f(-pi)
ans =
   -0.0983
```

**Comment by mhsnjvd** [comment]
@trefethen The only file I touched in the fix #2051 was ``truncate.m``, so I don't understand how this might have effected the constructor.

The code you have doesn't declare ``f`` via a function handle or a string: 
```
>> f = chebfun(exp(-10*x.^2),[-pi,pi],'trig');
```
However, when you write 
```
>> f = chebfun('exp(-10*x.^2)',[-pi,pi],'trig');
```
or 
```
>> f = chebfun(@(x) exp(-10*x.^2),[-pi,pi],'trig');
```
both work on development at my machine. 

Is this just a typo or am I missing something obvious?

**Comment by trefethen** [comment]
@mhsnjvd, thank you!  I did indeed make exactly this mistake (having no doubt earlier defined |x| as a chebfun on [-1,1]).  Mystery resolved.  Sorry for wasting your time.  I will close this again.
